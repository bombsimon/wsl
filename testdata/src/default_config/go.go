package testpkg

import "fmt"

func Go() {
	fooFunc := func() {}
	go fooFunc()

	barFunc := func() {}
	go fooFunc() // want "missing whitespace decreases readability"

	go func() {
		fmt.Println("hey")
	}()

	cuddled := true
	go func() { // want "missing whitespace decreases readability"
		fmt.Println("hey")
	}()

	argToGo := 1
	go takesArg(argToGo)

	notArgToGo := 1
	go takesArg(argToGo) // want "missing whitespace decreases readability"

	t1 := NewT()
	t2 := NewT()
	t3 := NewT()

	go t1()
	go t2()
	go t3()

	multiCuddle1 := NewT()
	multiCuddle2 := NewT() // want "missing whitespace decreases readability"
	go multiCuddle2()

	// TODO: Breaking change, this used to be on the first `go` stmt - now it's
	// on the line that should have a blank line above.
	t4 := NewT()
	t5 := NewT() // want "missing whitespace decreases readability"
	go t5()
	go t4()
}
