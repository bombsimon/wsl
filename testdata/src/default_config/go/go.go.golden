
package testpkg

import "fmt"

func NewT() func() {
	return func() {}
}

func Fn(_ int) {}

func Go() {
	fooFunc := func() {}
	go fooFunc()

	barFunc := func() {}

	go fooFunc() // want `missing whitespace above this line \(no shared variables above go\)`

	_ = barFunc

	go func() {
		fmt.Println("hey")
	}()

	cuddled := true

	go func() { // want `missing whitespace above this line \(no shared variables above go\)`
		fmt.Println("hey")
	}()

	_ = cuddled

	argToGo := 1
	go Fn(argToGo)

	notArgToGo := 1

	go Fn(argToGo) // want `missing whitespace above this line \(no shared variables above go\)`

	_ = notArgToGo

	t1 := NewT()
	t2 := NewT()
	t3 := NewT()

	go t1()
	go t2()
	go t3()

	multiCuddle1 := NewT()

	multiCuddle2 := NewT() // want `missing whitespace above this line \(too many statements above go\)`
	go multiCuddle2()

	t4 := NewT()

	t5 := NewT() // want `missing whitespace above this line \(too many statements above go\)`
	go t5()
	go t4()

	_ = t1
	_ = t2
	_ = t3
	_ = t4
	_ = t5
	_ = multiCuddle1
	_ = multiCuddle2
}
